src/main/java/com/exasol/adapter/request/parser/PushdownSqlParser.java

```diff
@@ -182,40 +181,93 @@ private SqlNode parseLiteralBool(final JsonObject expression) {
     }
 
     private List<SqlNode> parseExpressionList(final JsonArray array) {
-        assert array != null;
         final List<SqlNode> sqlNodes = new ArrayList<>();
         for (final JsonObject expr : array.getValuesAs(JsonObject.class)) {
-            final SqlNode node = parseExpression(expr);
-            sqlNodes.add(node);
+            sqlNodes.add(parseExpression(expr));
         }
         return sqlNodes;
     }
 
     private SqlGroupBy parseGroupBy(final JsonArray groupBy) {
-        if (groupBy == null) {
-            return null;
-        }
-        final List<SqlNode> groupByElements = parseExpressionList(groupBy);
-        return new SqlGroupBy(groupByElements);
+        return groupBy == null ? null : new SqlGroupBy(parseExpressionList(groupBy));
     }
 
     private SqlNode parseLiteralDate(final JsonObject exp) {
         final String date = exp.getString(VALUE);
         return new SqlLiteralDate(date);
     }
 
-    private SqlSelectList parseSelectList(final JsonArray selectList) {
-        if (selectList == null) {
-            return SqlSelectList.createSelectStarSelectList();
-        }
-        final List<SqlNode> selectListElements = parseExpressionList(selectList);
+    private SqlSelectList parseSelectList(final JsonArray selectList, final SqlNode from) {
+        final List<SqlNode> selectListElements = (selectList == null) ? collectAllInvolvedColumns(from)
+                : parseExpressionList(selectList);
```

Comment:
I prefer the old if-structure here, because `collectAllInvolvedColumns()` should never return an empty list, so we can return immediately in that case.

Also, if the database unintentionally sends `involvedTables` with a table with an empty column list, the parser never notices this. And then we reach this point with an empty select list. I think the parser should prevent such empty column lists, instead of generating SELECT TRUE here.